examples/meta/src/neural_nets/feedforward_net_classification.sg

```diff
@@ -1,28 +1,31 @@
-CSVFile f_feats_train("../../data/classifier_binary_2d_nonlinear_features_train.dat")
-CSVFile f_feats_test("../../data/classifier_binary_2d_nonlinear_features_test.dat")
-CSVFile f_labels_train("../../data/classifier_binary_2d_nonlinear_labels_train.dat")
-CSVFile f_labels_test("../../data/classifier_binary_2d_nonlinear_labels_test.dat")
+File f_feats_train = csv_file("../../data/classifier_binary_2d_nonlinear_features_train.dat")
+File f_feats_test = csv_file("../../data/classifier_binary_2d_nonlinear_features_test.dat")
+File f_labels_train = csv_file("../../data/classifier_binary_2d_nonlinear_labels_train.dat")
+File f_labels_test = csv_file("../../data/classifier_binary_2d_nonlinear_labels_test.dat")
 
 Math:init_random(1)
 
 #![create_features]
 Features features_train = features(f_feats_train)
 Features features_test = features(f_feats_test)
-BinaryLabels labels_train(f_labels_train)
-BinaryLabels labels_test(f_labels_test)
+Labels labels_train = labels(f_labels_train)
+Labels labels_test = labels(f_labels_test)
 #![create_features]
 
 #![add_layers]
 int num_feats = features_train.get_int("num_features")
-NeuralLayers layers() 
-layers.input(num_feats)
-layers.rectified_linear(10)
-layers.softmax(2)
-DynamicObjectArray all_layers = layers.done()
+DynamicObjectArray all_layers()
+NeuralLayer input = neural_layer("NeuralInputLayer", num_neurons=num_feats)
+all_layers.append_element(input)
+NeuralLayer relu = neural_layer("NeuralRectifiedLinearLayer", num_neurons=10)
+all_layers.append_element(relu)
+NeuralLayer softmax = neural_layer("NeuralSoftmaxLayer", num_neurons=2)
+all_layers.append_element(softmax)
 #![add_layers]
 
 #![create_instance]
-NeuralNetwork network(all_layers)
+NeuralNetwork network = neural_network("NeuralNetwork", labels=labels_train)
```

Comment:
Neural network is a machine, so this could be instantiated using `machine`
This means that the below methods `set_layers`, `quick_connect`, and `initialize_neural_network` all would have to be changed.
* `set_layers` can be done via put (already now!) just needs a dispatcher for the `CDynamicObjectArray` in put.
* `quick_connect`, and `initialize_neural_network` could just be called automatically inside the `train_machine` method which should be a minor refactor